sbt-tasty-mima/src/main/scala/sbttastymima/TastyMiMaPlugin.scala

```diff
@@ -0,0 +1,197 @@
+        }
+        val module = lm.wrapDependencyInModule(artifactID, scalaModInfo)
+
+        lm.update(module, updateConfiguration, UnresolvedWarningConfiguration(), log) match {
+          case Left(unresolvedWarning) =>
+            throw unresolvedWarning.resolveException
+
+          case Right(updateReport) =>
+            val cp0: Seq[(ModuleID, Path)] = for {
+              config <- updateReport.configurations
+              m <- config.modules
+              (artifact, file) <- m.artifacts
+            } yield {
+              m.module -> file.toPath()
+            }
+            val cp: Seq[Path] = javaBootCp ++ cp0.map(_._2)
+
+            val entry: Path = cp0.find { pair =>
+              pair._1 == artifactID
+            }.getOrElse {
+              throw new MessageOnlyException(s"Could not find entry for $artifactID in " + cp0.mkString("\n", "\n", ""))
+            }._2
+
+            (artifactID, cp, entry)
+        }
+      }
+    },
+    tastyMiMaCurrentClasspath := {
+      val javaBootCp = tastyMiMaJavaBootClasspath.value
+      val classDir = (Compile / classDirectory).value.toPath()
+      val jar = (Compile / packageBin / artifactPath).value.toPath()
+      val cp0 = Attributed.data((Compile / fullClasspath).value).map(_.toPath())
+
+      val cp: Seq[Path] = javaBootCp ++ cp0
+      val entry: Path = cp0.find { path =>
+        path == classDir || path == jar
+      }.getOrElse {
+        throw new MessageOnlyException(s"Could not find entry $classDir or $jar in " + cp0.mkString("\n", "\n", ""))
+      }
+
+      (cp, entry)
+    },
+    tastyMiMaReportIssues := {
+      val projID = projectID.value
+      val log = streams.value.log
+
+      val tastyMiMaCp = Attributed.data(tastyMiMaClasspath.value).map(_.toURI().toURL()).toArray
+      val config = tastyMiMaConfig.value
+      val tastyMiMa = tastymima.intf.TastyMiMa.newInstance(tastyMiMaCp, getClass().getClassLoader(), config)
+
+      val previousCps = tastyMiMaPreviousClasspaths.value
+      val currentCp = tastyMiMaCurrentClasspath.value
+
+      analyzeAndReport(projID, log, tastyMiMa, previousCps, currentCp)
+    },
+  )
+
+  private def analyzeAndReport(
+    projectID: ModuleID,
+    log: Logger,
+    tastyMiMa: tastymima.intf.TastyMiMa,
+    previousCps: Seq[(ModuleID, Seq[Path], Path)],
+    currentCp: (Seq[Path], Path),
+  ): Unit = {
+    import scala.collection.JavaConverters.*
+
+    val newClasspath = currentCp._1.asJava
+    val newClasspathEntry = currentCp._2
+
+    var anyFailed: Boolean = false
+
+    for (previousCp <- previousCps) {
+      val previousID = previousCp._1
+      val oldClasspath = previousCp._2.asJava
+      val oldClasspathEntry = previousCp._3
+
+      val problems = tastyMiMa.analyze(oldClasspath, oldClasspathEntry, newClasspath, newClasspathEntry).asScala
+
+      if (problems.nonEmpty) {
+        log.error(s"TASTy compatibility check failed for $projectID with respect to $previousID")
```

Comment:
It could be a bit confusing that e.g. `previousID` includes the platform/scalaversion suffix, and `projectID` does not

Reply:
possibly just have the project name and version?